x-pack/plugins/uptime/server/rest_api/pings/get_pings.ts

```diff
@@ -5,37 +5,41 @@
  */
 
 import { schema } from '@kbn/config-schema';
+import { isLeft } from 'fp-ts/lib/Either';
+import { PathReporter } from 'io-ts/lib/PathReporter';
 import { UMServerLibs } from '../../lib/lib';
 import { UMRestApiRouteFactory } from '../types';
 import { API_URLS } from '../../../../../legacy/plugins/uptime/common/constants/rest_api';
+import { GetPingsParamsType } from '../../../../../legacy/plugins/uptime/common/runtime_types';
 
 export const createGetPingsRoute: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
   method: 'GET',
   path: API_URLS.PINGS,
   validate: {
     query: schema.object({
-      dateRangeStart: schema.string(),
-      dateRangeEnd: schema.string(),
+      from: schema.string(),
+      to: schema.string(),
       location: schema.maybe(schema.string()),
       monitorId: schema.maybe(schema.string()),
+      index: schema.maybe(schema.number()),
       size: schema.maybe(schema.number()),
       sort: schema.maybe(schema.string()),
       status: schema.maybe(schema.string()),
     }),
   },
   handler: async ({ callES, dynamicSettings }, _context, request, response): Promise<any> => {
-    const { dateRangeStart, dateRangeEnd, location, monitorId, size, sort, status } = request.query;
+    const { from, to, ...optional } = request.query;
+    const params = GetPingsParamsType.decode({ dateRange: { from, to }, ...optional });
+    if (isLeft(params)) {
+      // eslint-disable-next-line no-console
+      console.error(new Error(PathReporter.report(params).join(';')));
+      return response.badRequest({ body: { message: 'Received invalid request parameters.' } });
+    }
```

Comment:
this is also unnecessary, since kibana route schema validation does it auto for all APIs.

Reply:
👍 I only find it nicer to have typing on the parameters but I agree it's redundant. Wish Kibana schema caused the params to be typed instead of `any`.

I've already removed it in any case.